src/chttpd.erl

```diff
@@ -209,21 +204,58 @@ handle_request_int(MochiReq) ->
                 || Part <- string:tokens(Path, "/")]
     },
 
-    {ok, HttpReq} = chttpd_plugin:before_request(HttpReq0),
+    % put small token on heap to keep requests synced to backend calls
+    erlang:put(nonce, Nonce),
+
+    % suppress duplicate log
+    erlang:put(dont_log_request, true),
+
+    Result0 = case before_request(HttpReq0) of
+        {ok, HttpReq1} ->
+            process_request(HttpReq1);
+        {error, Response} ->
+            {HttpReq0, Response}
+    end,
+
+    {HttpReq3, HttpResp0} = result(Result0, HttpReq0),
```

Comment:
The way how we form Result0 and how result/2 works very confusing. Especially for the case when before_request returns {error, Response}, because Response have to be {ok, #httpd{}}  or {aborted, MochiReq, Reason} or you'll get function_clause.

Is it possible to make it more friendly for this refactor?